### PR TITLE
Chapter03 - Jakarta EE 10 Core Profile

### PR DESCRIPTION
**Pull request by ttelang**
Replacement PR for #8

### CODE REVIEW

File: chapter03/chapter03.adoc [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+The Jakarta EE 10 Core Profile is a streamlined subset of the full Jakarta EE platform explicitly designed for building lightweight microservices and cloud-native applications. It provides a standardized foundation for smaller runtime environments, comprising of a curated selection of Jakarta EE specifications:
+
+* *Jakarta Annotations*: Enables developers to decorate their code with metadata to influence system configuration and behavior, making the code concise, readable, and maintainable.
+
+* *Jakarta Contexts and Dependency Injection Lite*: Facilitates the management of lifecycle contexts of stateful components and the injection of dependencies.
+
+* *Jakarta Interceptors*: Offers a means to intercept business method invocations and lifecycle events, ideal for implementing cross-cutting concerns such as logging.
+
+* *Jakarta JSON*: Simplifies the parsing, generation, and binding of JSON data for Java objects, crucial for RESTful service communication.
```
Comment (Emily-Jiang) [inline chapter03/chapter03.adoc:33]: This should mention JSON Processing and JSON Binding

Comment (ttelang) [reply chapter03/chapter03.adoc]: Fixed.

**Comment by ttelang** [comment]
@Emily-Jiang - Please merge this PR.

File: chapter03/chapter03.adoc [hunk 2]
```diff
@@ -0,0 +1,504 @@
+= Chapter 03: Jakarta EE 10 Core Profile
+
+== Introduction
+
+This chapter delves into the *Jakarta EE 10 Core Profile*, a specification designed specifically for microservices and cloud-native apps. *Jakarta EE* is a comprehensive framework within the Java ecosystem for crafting enterprise-grade applications. Complementing this, *MicroProfile* addresses the intricacies of microservices development, such as configuration management, fault tolerance, health checks, and monitoring. The foundation of MicroProfile is built on the strong and established standards of Jakarta EE, which ensures smooth integration of these modern APIs with the enterprise Java landscape.
+
+In this chapter, through practical examples, we will explore the critical features of the Jakarta EE 10 Core Profile that are most relevant to microservices development, including Contexts and Dependency Injection (CDI), Jakarta REST (formerly JAX-RS), JSON Binding and JSON Processing.
```
Comment (Emily-Jiang) [inline chapter03/chapter03.adoc:7]:
```suggestion
In this chapter, through practical examples, we will explore the critical features of the Jakarta EE 10 Core Profile that are most relevant to microservices development, including Contexts and Dependency Injection (CDI), Jakarta RESTful Web Services (Jakarta REST, formerly JAX-RS), JSON Binding and JSON Processing.
```

File: chapter03/chapter03.adoc [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+Integration with JSON-P: JSON-B is built on top of JSON-P and can seamlessly integrate with it, allowing developers to mix high-level object binding with low-level JSON processing as needed.
+
+NOTE: If you are interested in diving deeper into the specifics of JSON Binding, We highly recommend you to visit the Jakarta EE tutorial. It provides detailed insights into how JSON Binding works, including the processes for converting Java objects to JSON and vice versa. This knowledge is crucial for effectively managing JSON data in Java-based enterprise applications. Learn more at the link:https://jakartaee.github.io/jakartaee-documentation/jakartaee-tutorial/current/web/jsonb/jsonb.html[Jakarta EE Documentation on JSON Binding].
+
+=== Jakarta RESTful Web Services
+
+Jakarta RESTful Web Services is a specification for creating web services according to the Representational State Transfer (REST) architectural pattern. It provides annotations to define resources and operations, making it straightforward to develop APIs for web applications.
```
Comment (Emily-Jiang) [inline chapter03/chapter03.adoc:115]:
```suggestion
Jakarta RESTful Web Services(Jakarta REST) is a specification for creating web services according to the Representational State Transfer (REST) architectural pattern. It provides annotations to define resources and operations, making it straightforward to develop APIs for web applications.
```

File: chapter03/chapter03.adoc [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+
+CDI is the specification that unifies the Jakarta EE platform by providing a consistent way to manage the lifecycle of stateful components and their interactions. The CDI Lite section of the specification is tailored for environments where full CDI support may be too heavyweight, such as microservices and serverless deployments.
+
+==== Key Features of Contexts and Dependency Injection (CDI) - CDI Lite
+
+* *Type-safe Dependency Injection*: Enables the injection of beans in a type-safe manner, reducing runtime errors and improving developer productivity.
+
+* *Contextual Lifecycle Management*: Manages the lifecycle of beans according to well-defined contexts, simplifying state management across different scopes.
+
+* *Interceptors and Decorators*: Supports the use of interceptors and decorators for adding behavior to beans or for altering their behavior in a non-invasive manner.
```
Comment (Emily-Jiang) [inline chapter03/chapter03.adoc:63]:
```suggestion
* *Interceptors*: Supports the use of interceptors for adding behavior to beans or for altering their behavior in a non-invasive manner.
```

Comment (Emily-Jiang) [reply chapter03/chapter03.adoc:63]: Decorator is NOT part of CDI lite.

**Comment by Emily-Jiang** [comment]
@ttelang please my above comments

**Comment by ttelang** [comment]
> @ttelang please address my above comments

Thanks, Any specific reason why you are preferring Jakarta REST in parenthesis like "Jakarta RESTful Web Services (Jakarta REST)"

**Comment by Emily-Jiang** [comment]
> > @ttelang please address my above comments
> 
> Thanks, Any specific reason why you are preferring Jakarta REST in parenthesis like "Jakarta RESTful Web Services (Jakarta REST)"

Jakarta RESTful Web Services is the official name while Jakarta REST is the acronym.

**Review by Emily-Jiang — APPROVED** [review]
LGTM

**Pull request merged**